tests/integration/diag_scripts/mlr/_sklearn_utils.py

```diff
@@ -0,0 +1,295 @@
+        result = func(*args, **kw)
+        # Verify some things
+        if not len(warn) > 0:
+            raise AssertionError("No warning raised when calling %s"
+                                 % func.__name__)
+
+        found = [issubclass(warning.category, warning_class) for warning in
+                 warn]
+        if not any(found):
+            raise AssertionError("No warning raised for %s with class "
+                                 "%s"
+                                 % (func.__name__, warning_class))
+
+        message_found = False
+        # Checks the message of all warnings belong to warning_class
+        for index in [i for i, x in enumerate(found) if x]:
+            # substring will match, the entire message with typo won't
+            msg = warn[index].message  # For Python 3 compatibility
+            msg = str(msg.args[0] if hasattr(msg, 'args') else msg)
+            if callable(message):  # add support for certain tests
+                check_in_message = message
+            else:
+                def check_in_message(msg):
+                    return message in msg
+
+            if check_in_message(msg):
+                message_found = True
+                break
+
+        if not message_found:
+            raise AssertionError("Did not receive the message you expected "
+                                 "('%s') for <%s>, got: '%s'"
+                                 % (message, func.__name__, msg))
+
+    return result
+
+
+def assert_raise_message(exceptions, message, function, *args, **kwargs):
+    """Test whether the message is in an exception.
+
+    Given an exception, a callable to raise the exception, and
+    a message string, tests that the correct exception is raised and
+    that the message is a substring of the error thrown. Used to test
+    that the specific message thrown during an exception is correct.
+
+    Parameters
+    ----------
+    exceptions : exception or tuple of exception
+        An Exception object.
+
+    message : str
+        The error message or a substring of the error message.
+
+    function : callable
+        Callable object to raise error.
+
+    *args : the positional arguments to `function`.
+
+    **kwargs : the keyword arguments to `function`.
+
+    """
+    try:
+        function(*args, **kwargs)
+    except exceptions as exc:
+        error_message = str(exc)
+        if message not in error_message:
+            raise AssertionError("Error message does not include the expected "
+                                 "string: %r. Observed error message: %r" %
+                                 (message, error_message))
+    else:
+        # concatenate exception names
+        if isinstance(exceptions, tuple):
+            names = " or ".join(e.__name__ for e in exceptions)
+        else:
+            names = exceptions.__name__
+
+        raise AssertionError("%s not raised by %s" %
+                             (names, function.__name__))
```

Comment:
same for these two funcs - consider including them upstream so they can be used by others?

Reply:
See comment above